engines/base/src/main/java/org/alfresco/transform/base/probes/ProbeTransform.java

```diff
@@ -0,0 +1,372 @@
+
+    private final boolean livenessTransformEnabled;
+    private final long livenessTransformPeriod;
+    private final long maxTransformCount;
+    private long maxTransformTime;
+
+    private final AtomicBoolean initialised = new AtomicBoolean(false);
+    private final AtomicBoolean readySent = new AtomicBoolean(false);
+    private final AtomicLong transformCount = new AtomicLong(0);
+    private final AtomicBoolean die = new AtomicBoolean(false);
+
+    public int getLivenessPercent()
+    {
+        return livenessPercent;
+    }
+
+    public long getMaxTime()
+    {
+        return maxTime;
+    }
+
+    public ProbeTransform(String sourceFilename, String sourceMimetype, String targetMimetype, Map<String, String> transformOptions,
+        long expectedLength, long plusOrMinus, int livenessPercent, long maxTransforms, long maxTransformSeconds,
+        long livenessTransformPeriodSeconds)
+    {
+        this.sourceFilename = sourceFilename;
+        this.sourceMimetype = sourceMimetype;
+        this.targetMimetype = targetMimetype;
+        this.transformOptions = new HashMap(transformOptions);
+        minExpectedLength = Math.max(0, expectedLength - plusOrMinus);
+        maxExpectedLength = expectedLength + plusOrMinus;
+
+        this.livenessPercent = (int) getPositiveLongEnv("livenessPercent", livenessPercent);
+        maxTransformCount = getPositiveLongEnv("maxTransforms", maxTransforms);
+        maxTransformTime = getPositiveLongEnv("maxTransformSeconds", maxTransformSeconds) * 1000;
+        livenessTransformPeriod = getPositiveLongEnv("livenessTransformPeriodSeconds",
+            livenessTransformPeriodSeconds) * 1000;
+        livenessTransformEnabled = getBooleanEnvVar("livenessTransformEnabled", false);
+    }
+
+    private boolean getBooleanEnvVar(final String name, final boolean defaultValue)
+    {
+        try
+        {
+            return Boolean.parseBoolean(System.getenv(name));
+        }
+        catch (Exception ignore)
+        {
+        }
+        return defaultValue;
+    }
+
+    private long getPositiveLongEnv(String name, long defaultValue)
+    {
+        long l = -1;
+        String env = System.getenv(name);
+        if (env != null)
+        {
+            try
+            {
+                l = Long.parseLong(env);
+            }
+            catch (NumberFormatException ignore)
+            {
+            }
+        }
+        if (l <= 0)
+        {
+            l = defaultValue;
+        }
+        logger.trace("Probe: {}={}", name, l);
+        return l;
+    }
+
+    // We don't want to be doing test transforms every few seconds, but do want frequent live probes.
+    public String doTransformOrNothing(HttpServletRequest request, boolean isLiveProbe, TransformHandler transformHandler)
+    {
+        // If not initialised OR it is a live probe and we are scheduled to to do a test transform.
+        probeCount++;
+        // TODO: update/fix/refactor liveness probes as part of ATS-138
```

Comment:
I think we will need to look at probes again anyway in HxP as I discovered this code had been cobbled since I last looked at it. Keep the ref for now.